lib/plugins/blocks.js

```diff
@@ -227,6 +227,77 @@ function inject (bot, { version, storageBuilder }) {
     return bot.blockAt(blocks[0])
   }
 
+  bot.findBlocks2 = async (opts) => {
+    assert(
+      typeof opts.wantedBlock === 'number' || typeof opts.wantedBlocks === 'object',
+      Error('wantedBlock has to be an id of a block or wantedBlocks has to be an array of block ids')
+    )
+    assert(
+      typeof opts.wantedBlock !== 'number' || typeof opts.wantedBlocks !== 'object',
+      Error('Either set wantedBlock or wantedBlocks, not both')
+    )
+    assert(
+      typeof opts.matcher === 'function' || opts.matcher === undefined,
+      Error('Matcher must be a function')
+    )
+    assert(
+      opts.maxMatches > 0 || opts.maxMatches === undefined,
+      Error('Max blocks must be >0')
+    )
+    assert(
+      opts.searchRadius > 0 || opts.searchRadius === undefined,
+      Error('Search Radius must be >0')
+    )
+    assert(
+      opts.point instanceof Vec3 || opts.point === undefined,
+      Error('Start point must be a vec3 point')
+    )
+    const internalWantedBlocks = opts.wantedBlocks || [opts.wantedBlock]
+    const searchRadius = opts.searchRadius || 16
+
+    const foundBlocks = []
+    const { x: originX, y: originY, z: originZ } = (opts.point || bot.entity.position).floored()
+    // start and end positions
+    const startX = originX - searchRadius
+    let startY = originY - searchRadius
+    const startZ = originZ - searchRadius
+    const endX = originX + searchRadius
+    let endY = originY + searchRadius
+    const endZ = originZ + searchRadius
+    // if set by user, change start and end pos
+    if (opts.toBedrock) startY = 0
+    if (opts.toSkyLimit) endY = 256
+    // find blocks
+    for (let x = startX; x < endX; x++) {
+      for (let y = startY; y < endY; y++) {
+        for (let z = startZ; z < endZ; z++) {
+          const currBlock = new Vec3(x, y, z)
+          foundBlocks.push(currBlock)
+        }
+      }
+    }
+    // filter for wanted blocks
+    let matcher = opts.matcher
+    if (matcher === undefined) {
+      matcher = async (currentBlock) => {
+        const { type } = await bot.world.getBlock(currentBlock)
+        return internalWantedBlocks.includes(type)
+      }
+    }
+    let wantedBlocks = foundBlocks.filter(matcher)
+    // sort blocks by distance from bot
+    const botLocation = bot.entity.position.floored()
```

Comment:
If you really wanted to have an array with all the positions, you could have sorted it before doing the matching, so you could stop when reaching the count.